ReportPortal.Tests/ReportPortal.E2E.API.Tests/Scenarios/MsTest/LaunchesQueryByUserMsTestTests.cs

```diff
@@ -0,0 +1,48 @@
+﻿using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using ReportPortal.E2E.API.Business;
+using ReportPortal.E2E.API.Business.Models.Responses;
+using ReportPortal.E2E.API.Tests.Scenarios.MsTest.BaseTest;
+using ReportPortal.E2E.Core.Extensions;
+using ReportPortal.E2E.Core.Models.TestDataModel;
+using ReportPortal.E2E.Core.Utility;
+
+namespace ReportPortal.E2E.API.Tests.Scenarios.MsTest
+{
+    [TestClass]
+    public class LaunchesQueryByUserMsTestTests : MsTestBaseTest
+    {
+        [TestMethod]
+        [DataTestMethod]
+        [DynamicData(nameof(GetLaunchesQuery), DynamicDataSourceType.Method)]
+        public void Filter_Demo_Launches_By_Query(string query, Int64 count)
+        {
+            var launchesList = Steps.AsUser(NewUserCredentials).GetLaunchesByFilter(ProjectName, query).GetAwaiter().GetResult()
+                .GetResponse<GetLaunchesResponse>().Launches;
+            launchesList.Should().HaveCount((int)count);
+        }
```

Comment:
Why do you convert from Int64 to int? Is it possible to use int from start?

Reply:
Yes, this is possible, but you need to keep in mind that if you do not explicitly indicate the type of the variable, and parse the JSON file, by default the variable type is assigned to long(int64). All the options that I have found so that an array of objects can contain both strings and numbers at the same time is:
1) you need to iterate over all the elements and parse only those that can be parsed.
2) accept a number as a string in the test itself and parse from string to number already in the test.

My solution: I decided that it would be much easier to use int64 by default, because... Perhaps sooner or later we can start using long variables in the test data (for example, long is used to indicate the start time of the launch, and also to indicate its end). Please let me know if I missed something in my judgment.

Reply:
Resolved